launch_ros/launch_ros/actions/node.py

```diff
@@ -337,7 +337,7 @@ def _perform_substitutions(self, context: LaunchContext) -> None:
                 validate_node_name(self.__expanded_node_name)
             self.__expanded_node_name.lstrip('/')
             expanded_node_namespace: Optional[Text] = None
-            if self.__node_namespace:
+            if self.__node_namespace is not None:
```

Comment:
doesn't [this comment](https://github.com/ros2/launch_ros/pull/179#discussion_r492017050) still apply?

Reply:
IIUC, that comment is concerned about the case when the node sets a namespace in code. I tried doing this and it behaves the same before and after the change in this PR: if a namespace is given in the launch file then the node uses that namespace, otherwise it will use the namespace set in the node.

So, I guess not?

Reply:
err, I guess the comment does still apply. I see differences if I pass an explicitly empty namespace in launch. The behavior was broken in #179. Prior to #179, `namespace=""` uses the namespace set in code and now it uses `/`...

Reply:
However, this is the same behavior on Rolling. Assuming this is the desired behavior, and we already released this behavior into Foxy, then I'm inclined to leave it as-is. What do you think?

Reply:
I agree that having the same behavior that `Rolling` is nice, but it might break some previous launch files so IDK what's best.

Reply:
If it's going to break launch files in Foxy, then it already has since #179 was released a while ago. I think this PR isn't breaking an existing launch file unless they were depending on the bug described in #220.